Src/GoogleApis.Auth.DotNet4/OAuth2/GoogleCredential.cs

```diff
@@ -0,0 +1,307 @@
+        /// </para>
+        /// </summary>
+        public static GoogleCredential FromStream(Stream stream)
+        {
+            return defaultCredentialProvider.CreateDefaultCredentialFromStream(stream);
+        }
+
+        /// <summary>
+        /// <para>Returns true only if this credential type has no scopes by default and requires 
+        /// a call to createScoped before use.</para>
+        ///
+        /// <para>Credentials need to have scopes in them before they can be used to access Google services. 
+        /// Some Credential types have scopes built-in, and some dont. This property indicates whether 
+        /// the Credential type has scopes built-in.</para>
+        /// 
+        /// <list type="number">
+        /// <item> 
+        /// <description>ComputeCredential has scopes built-in. Nothing additional is required.</description> 
+        /// </item> 
+        /// <item> 
+        /// <description>UserCredential has scopes built-in, as they were obtained during the consent 
+        /// screen. Nothing additional is required.</description> 
+        /// </item> 
+        /// <item> 
+        /// <description>ServiceAccountCredentials does not have scopes built-in by default. Caller should 
+        /// invoke <see cref="Google.Apis.Auth.OAuth2.IScopableCredential.CreateScoped"/> to add scopes to the Credential.</description> 
+        /// </item> 
+        /// </list>
+        /// </summary>
+        public virtual bool IsCreateScopedRequired
+        {
+            get { return false; }
+        }
+
+        /// <summary>If the credential supports scopes, creates a copy with the specified scopes, otherwise return the same instance.</summary>
+        public virtual GoogleCredential CreateScoped(IEnumerable<string> scopes)
+        {
+            return this;
+        }
+
+        #region IConfigurableHttpClientInitializer
+
+        void IConfigurableHttpClientInitializer.Initialize(ConfigurableHttpClient httpClient)
+        {
+            Initialize(httpClient);
+        }
+
+        #endregion
+
+        #region ITokenAccess
+
+        TokenResponse ITokenAccess.Token
+        {
+            get { return Token; }
+        }
+
+        Task<bool> ITokenAccess.RequestAccessTokenAsync(CancellationToken taskCancellationToken)
+        {
+            return RequestAccessTokenAsync(taskCancellationToken);
+        }
+
+        #endregion
+
+        /// <summary>Provides access to the underlying credential object</summary>
+        internal abstract object UnderlyingCredential { get; }
+
+        // We're explicitly implementing all the interfaces to only expose the members user actually
+        // needs to see. Because you cannot make explicit interface implementors abstract, they are redirecting
+        // to the following protected abstract members.
+        protected abstract void Initialize(ConfigurableHttpClient httpClient);
```

Comment:
This note doesn't belong to the method itself but to the following three abstract members. In C#, it is illegal to explicitly implement an interface (which is what I want to do here) and leave those members as abstract. So I explicitly implement ITokenAccess.Token for example and redirect it to a protected abstract member (that the subclasses will implement).

This is an implementation detail that will go away once I address the TODO to make all credential classes implement the ITokenAccess interface. Let's leave it for now.